src/coreclr/vm/pendingload.h

```diff
@@ -17,248 +17,404 @@
 #include "shash.h"
 #include "typestring.h"
 
+#ifdef PENDING_TYPE_LOAD_TABLE_STATS
+// Enable PENDING_TYPE_LOAD_TABLE_STATS to gather counts of Entry structures which are allocated
+void PendingTypeLoadEntryDynamicAlloc();
+#endif // PENDING_TYPE_LOAD_TABLE_STATS
+
 //
 // A temporary structure used when loading and resolving classes
 //
-class PendingTypeLoadEntry
+
+// Hash table used to hold pending type loads
+// This is a sharded fixed bucket count hashtable, with locking per shard
+// This is a singleton
+class PendingTypeLoadTable
 {
-    friend class ClassLoader;       // workaround really need to beef up the API below
+    static void CallCrstEnter(Crst* pCrst)
+    {
+        pCrst->Enter(INDEBUG(Crst::CRST_NO_LEVEL_CHECK));
```

Comment:
What is the reason for the `CRST_NO_LEVEL_CHECK`?

Reply:
I assume this is from:
https://github.com/dotnet/runtime/blob/971ebdbd0fe335c34d440269af8055c6eea87286/src/coreclr/vm/pendingload.h#L54-L60

But I don't think that reasoning really applies to the refactored/static function now?